src/components/Tags.tsx

```diff
@@ -20,122 +25,114 @@ const KeyCodes = {
+        currentSegmentTags,
+        tags.map(reactTagToSegmentTag),
+        'tag.name'
+      );
+      // Keep segments of other ranks
+      currentSegmentTags = currentSegmentTags.filter(t => t.rank !== rank);
+      // Add new tags to old
+      currentSegmentTags = [...currentSegmentTags, ...tags.map(reactTagToSegmentTag)];
+      setReactTags(currentSegmentTags.filter(t => t.rank === rank).map(segmentTagToReactTag));
+      updateSegmentAt(segmentIndex, { tags: currentSegmentTags });
     }
-  }
+  };
 
-  handleAddition(tag: any) {
-    tag.id = uuid();
+  const handleDelete = (i: number): void => {
+    updateTags(reactTags.filter((tag, index) => index !== i));
+  };
 
+  const handleAddition = (tag: ReactTag): void => {
     //force lowercase
     tag.text = tag.text.toLowerCase();
+    updateTags([...reactTags, tag]);
+  };
 
-    //remove duplicates
-    for (let index = 0; index < this.props.seg.tags.length; index++) {
-      const element = this.props.seg.tags[index];
-      if (tag.text === this.props.seg.tags[index].tag.name) {
-        this.props.seg.tags.splice(index, 1);
-      }
-    }
-
-    this.setState(state => ({ tags: [...state.tags, tag] }));
-
-    let newtag: tagstruct = {
-      rank: this.props.rank,
-      segment: this.props.seg._id,
-      tag: { name: tag.text },
-      _id: uuid(),
-    };
-
-    this.props.seg.tags.push(newtag);
-    this.props.seg.pristine = true;
-    this.props.forceUpdate();
-  }
-
-  handleDrag(tag: any, currPos: any, newPos: any) {
-    const tags = [...this.state.tags];
-    const newTags = tags.slice();
+  const handleDrag = (tag: ReactTag, currPos: number, newPos: number) => {
+    const newTags = reactTags.slice();
 
     tag && newTags.splice(currPos, 1);
     tag && newTags.splice(newPos, 0, tag);
 
-    // re-render
-    this.setState({ tags: newTags });
-  }
+    updateTags(newTags);
+  };
 
-  handleTagClick(index: any) {
+  const handleTagClick = (index: number) => {
     console.log('The tag at index ' + index + ' was clicked');
-  }
-
-  render() {
-    const { tags, suggestions } = this.state;
-
-    return (
-      <div>
-        <ReactTags
-          tags={tags}
-          suggestions={suggestions}
-          delimiters={delimiters}
-          handleDelete={this.handleDelete}
-          handleAddition={this.handleAddition}
-          handleDrag={this.handleDrag}
-          handleTagClick={this.handleTagClick}
-          autofocus={false}
-          allowDeleteFromEmptyInput={false}
-          allowDragDrop={false}
-        />
-      </div>
-    );
-  }
-}
+  };
+
+  // This effect will run once during component initialization
+  React.useEffect(() => {
+    const tagsInput = currentSegment.tags || [];
+    if (assertModelArrayType<SegmentTag>(tagsInput, 'SegmentTag')) {
+      setReactTags(tagsInput.filter(tag => tag.rank === rank).map(segmentTagToReactTag));
+    }
+  }, []);
+
+  // This re-renders the tags when a refresh is triggered
+  React.useEffect(() => {
```

Comment:
`useEffect` hooks are very useful, and let you run a set of code whenever certain variables change (listed in the array second argument).

Here we check if the `refresh` property has been changed (which indicates that a different `Tags` component has updated) and update the tags for this component.